app/controllers/auth0_controller.rb

```diff
@@ -41,4 +72,39 @@ def request_password_reset
       redirect_to new_user_session_path
     end
   end
+
+  private
+
+  def filter_value(value, set_of_values)
+    value if set_of_values.include?(value)
+  end
+
+  def background_refresh_values
+    @auth0_token = auth0_decode_auth_token
+    if @auth0_token && @auth0_token[:auth_payload]
+      exp = @auth0_token[:auth_payload]["exp"]
+      iat = @auth0_token[:auth_payload]["iat"] || (exp - 15.minutes.in_seconds)
```

Comment:
could you add a comment in this block (e.g. the 15 minutes and the lifespan*0.2) ?

Reply:
That's a good observation. I'm adding comments and improving this logic.